src/Project.php

```diff
@@ -2669,6 +2669,266 @@ public function showDebug()
         NotificationEvent::debugEvent($this);
     }
 
+    /**
+     * Get the list of active projects for a list of groups.
+     *
+     * @param array $groups_id The group IDs.
+     * @param bool $search_in_team Whether to search in the team.
+     * @return array The list of project IDs.
+     */
+    public static function getActiveProjectIDsForGroup(
+        array $groups_id,
+        bool $search_in_team = true
+    ): array {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        $req = [
+            'SELECT' => Project::getTable() . '.id',
+            'FROM' => Project::getTable(),
+            'LEFT JOIN' => [
+                ProjectState::getTable() => [
+                    'FKEY' => [
+                        ProjectState::getTable() => 'id',
+                        Project::getTable() => 'projectstates_id'
+                    ]
+                ]
+            ],
+            'WHERE' => [
+                ['OR' => ['groups_id' => $groups_id]],
+                [
+                    'OR' => [
+                        [ProjectState::getTable() . '.is_finished' => 0],
+                        [ProjectState::getTable() . '.is_finished' => null]
+                    ]
+                ]
+            ]
+        ];
+
+        if ($search_in_team) {
+            $team_sub_query = new \Glpi\DBAL\QuerySubQuery([
+                'SELECT' => [
+                    'projects_id'
+                ],
+                'FROM' => ProjectTeam::getTable(),
+                'WHERE' => [
+                    'OR' => [
+                        ['itemtype' => 'Group', 'items_id' => $groups_id]
+                    ]
+                ]
+            ]);
+
+            $req['WHERE'][0]['OR'][Project::getTable() . '.id'] = $team_sub_query;
+        }
+
+        $iterator = $DB->request($req);
+
+        $project_ids = [];
+        foreach ($iterator as $data) {
+            $project_ids[] = $data['id'];
+        }
+
+        return $project_ids;
```

Comment:
```suggestion
        return iterator_to_array($iterator, false);
```

Can probably be simplified.